sdk/core/azure-core/test/ut/log_policy.cpp

```diff
@@ -0,0 +1,262 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include <azure/core/diagnostics/logger.hpp>
+#include <azure/core/http/policies/policy.hpp>
+#include <azure/core/internal/http/pipeline.hpp>
+
+#include <gtest/gtest.h>
+
+using Azure::Core::Diagnostics::Logger;
+using Azure::Core::Http::Policies::LogOptions;
+
+namespace {
+void SendRequest(LogOptions const& logOptions)
+{
+  using namespace Azure::Core;
+  using namespace Azure::Core::IO;
+  using namespace Azure::Core::Http;
+  using namespace Azure::Core::Http::_internal;
+  using namespace Azure::Core::Http::Policies;
+  using namespace Azure::Core::Http::Policies::_internal;
+
+  class TestTransportPolicy : public HttpPolicy {
+  public:
+    std::unique_ptr<HttpPolicy> Clone() const override
+    {
+      return std::make_unique<TestTransportPolicy>(*this);
+    }
+
+    std::unique_ptr<RawResponse> Send(Request&, NextHttpPolicy, Context const&) const override
+    {
+      constexpr uint8_t const responseBody[] = "Response Body";
+      constexpr uint8_t const responseBodyStream[] = "Request Body Stream";
```

Comment:
isn't it weird to define const expr here?  Consider making this static constexpr outside the method

Reply:
IDK, doesn't seem weird to me at all. I think we should use constexpr whenever possible, then const if constexpr is not possible, and then non-const if const is not possible. And we should use the most local scope, i.e. if `using`, class declaration, or const definition can be done inside a function, we should do it there.